### ISSUE TITLE
Exercise #4: Maybe level 5

### ISSUE THREAD

**Issue by GuoGangQuan** [nus-cs2030/2324-s1#72]
this my flatMap, it works but is there any better ways to do it?
![image](https://github.com/nus-cs2030/2324-s1/assets/147078222/9dd7c712-7848-4e8b-b101-16939642f513)

**Comment by wjunwei2001** [comment]
Another way is to assign mapper.apply(this.get()) to a Maybe<? extends U> variable first. Then afterwards .get() to retrieve the item inside and wrap it with Maybe<U>

Not sure if this method is better but it works

**Comment by jjiehh** [comment]
yep, i did the same thing as you did but instead of map(x -> x) i did map(Function.identity())

**Comment by yentheng0110** [comment]
Hello! Instead of using map, I did mapper.apply(this.get()).orElse(null) and then wrap it in a Maybe<R>. This method works
